content/stagingzone/worker.go

```diff
@@ -0,0 +1,234 @@
+						m.log.Warnf("failed to queue content: %d in staging zone queue for backfill - %s", cont.ID, err)
+						break
+					}
+				}
+
+				// move tracker forward
+				if err := m.db.Model(model.StagingZoneTracker{}).Where("id = ?", tracker.ID).UpdateColumn("last_cont_id", cont.ID).Error; err != nil {
+					m.log.Warnf("failed to move backfill stagingzone tracker - %s", err)
+					break
+				}
+			}
+
+			// if there are no more to backfill set stop
+			if len(contents) == 0 {
+				if err := m.db.Model(model.StagingZoneTracker{}).Where("id = ?", tracker.ID).UpdateColumn("stop_at", tracker.LastContID).Error; err != nil {
+					m.log.Warnf("failed to set stop_at for stagingzone queue tracker - %s", err)
+				}
+			}
+		}
+	}
+}
+
+func (m *manager) runCreationWorker(ctx context.Context) {
+	timer := time.NewTicker(m.cfg.WorkerIntervals.StagingZoneInterval)
+	for {
+		select {
+		case <-ctx.Done():
+			m.log.Info("shutting staging zone creation worker")
+			return
+		case <-timer.C:
+			m.log.Debug("running staging zone creation worker")
+
+			var tasks []*model.StagingZoneQueue
+			if err := m.db.Where("next_attempt_at < ?", time.Now().UTC()).Order("id asc").Limit(2000).Find(&tasks).Error; err != nil {
+				m.log.Warnf("failed to get staging zone tasks to stage - %s", err)
+				continue
+			}
+
+			for _, t := range tasks {
+				var cont *util.Content
+				if err := m.db.First(&cont, "id = ?", t.ContID).Error; err != nil {
+					if err != gorm.ErrRecordNotFound {
+						m.log.Warnf("failed to get content: %d for staging - %s", t.ContID, err)
+						break
+					} else {
+						m.log.Warnf("content: %d not found for staging - %s", t.ContID, err)
+						continue
+					}
+				}
+
+				var err error
+				if t.IsBackFilled {
+					err = m.stageBackfilledContent(ctx, cont)
+				} else {
+					err = m.stageNewContent(ctx, cont, cont.Size)
+				}
+
+				if err != nil {
+					m.log.Warnf("failed to stage content - %s", err)
+					if err = m.queueMgr.StageFailed(cont.ID, m.db); err != nil {
+						m.log.Warnf("failed to update staging queue - %s", err)
+					}
+				}
+			}
+		}
+	}
+}
+
+func (m *manager) runAggregationWorker(ctx context.Context) {
+	timer := time.NewTicker(m.cfg.WorkerIntervals.StagingZoneInterval)
+	for {
+		select {
+		case <-ctx.Done():
+			m.log.Info("shutting staging aggregation worker")
+			return
+		case <-timer.C:
+			m.log.Debug("running staging zone aggregation worker")
+
+			readyZones, err := m.getReadyStagingZones()
+			if err != nil {
+				m.log.Errorf("failed to get ready staging zones: %s", err)
+				continue
+			}
+
+			m.log.Debugf("found: %d ready staging zones, for aggregation", len(readyZones))
+
+			for _, z := range readyZones {
+				var zc *util.Content
+				if err := m.db.First(&zc, "id = ?", z.ContID).Error; err != nil {
+					m.log.Warnf("zone %d aggregation failed to get zone content %d for processing - %s", z.ID, z.ContID, err)
+					continue
+				}
+
+				err := m.processStagingZone(ctx, zc, z)
```

Comment:
Instead of passing the zone as a pointer, we should pass it as the value unless we need to update the reference at some point (which I don't see it being in done on this flow).

Reply:
ok, I will address this with the pinmgr refactor